orchagent/switchorch.cpp

```diff
@@ -431,14 +431,14 @@ void SwitchOrch::initSensorsTable()
             m_numTempSensors = attr.value.u8;
             m_numTempSensorsInitialized = true;
         }
-        else if (status ==  SAI_STATUS_NOT_SUPPORTED || status == SAI_STATUS_NOT_IMPLEMENTED)
+        else if (SAI_STATUS_IS_ATTR_NOT_SUPPORTED(status) || SAI_STATUS_IS_ATTR_NOT_IMPLEMENTED(status))
```

Comment:
The `SAI_STATUS_NOT_SUPPORTED` is for the request while `SAI_STATUS_IS_ATTR_NOT_SUPPORTED` is for the attribute.
In theory, `SAI_STATUS_IS_ATTR_NOT_SUPPORTED` is perfered but who knows whether other vendor will return `SAI_STATUS_NOT_SUPPORTED`.
Having said that, I don't have a strong opinion.
```
/**
 * @brief The request is not supported
 */
#define SAI_STATUS_NOT_SUPPORTED                    SAI_STATUS_CODE(0x00000002L)

```
```
/**
 * @brief Is attribute not supported helper
 */
#define SAI_STATUS_IS_ATTR_NOT_SUPPORTED(x)     (((x) & (~0xFFFF)) == SAI_STATUS_ATTR_NOT_SUPPORTED_0)

```